### ISSUE TITLE
Test fails after creating a project

### ISSUE THREAD

**Issue by m00g33k** [scaffold-js/generator-scaffold-js#13]
After I created a project and running "gulp serve", I keep on getting errors on tests which results to the server not unable to run. Here's the complete log from the terminal:
```

[06:11:06] Using gulpfile ~/Workspaces/testFrameworks/scaffold/gulpfile.js
[06:11:06] Starting 'js.lint'...
[06:11:06] Starting 'json.lint'...
[06:11:07] Starting 'test.unit.server.init'...
[06:11:07] Starting 'test.unit.web'...
[06:11:07] Starting 'nm'...
02 05 2016 06:11:07.503:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
02 05 2016 06:11:07.509:INFO [launcher]: Starting browser PhantomJS
[06:11:09] [nodemon] 1.9.2
[06:11:09] [nodemon] to restart at any time, enter `rs`
[06:11:09] [nodemon] watching: src/server/**/*.js
[06:11:09] [nodemon] starting `node server`
[06:11:09] Finished 'nm' after 1.94 s
[06:11:09] Starting 'bs'...
[06:11:09] Finished 'bs' after 28 ms
[Error: Configuration property "web.jwtKey" is not defined]
[06:11:10] [nodemon] app crashed - waiting for file changes before starting...
[BS] Proxying: http://localhost:8443
[BS] Access URLs:
 ------------------------------------
       Local: http://localhost:3000
    External: http://192.168.1.6:3000
 ------------------------------------
          UI: http://localhost:3001
 UI External: http://192.168.1.6:3001
 ------------------------------------
02 05 2016 06:11:12.024:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket /#HawpuBiAGsqAIK4ZAAAA with id 28082665
PhantomJS 2.1.1 (Linux 0.0.0): Executed 79 of 108 SUCCESS (0 secs / 2.238 secs)
[06:11:15] Finished 'test.unit.server.init' after 8.47 s
PhantomJS 2.1.1 (Linux 0.0.0): Executed 108 of 108 SUCCESS (2.654 secs / 3.129 secs)
[06:11:16] Finished 'test.unit.web' after 9.42 s


  server
    ✓ should start a server (86ms)
[Error: Configuration property "web.jwtKey" is not defined]
    ✓ should fail to start a server with an invalid plugin
[Error: Configuration property "web.jwtKey" is not defined]
    ✓ should fail to start a second server on the same port
[Error: Configuration property "web.jwtKey" is not defined]

  jwt helpers
    ✓ should validate valid token
    ✓ should fail to validate if host does not match
    ✓ should fail to validate if user not found
    ✓ should fail to validate if db fails

  server methods
    misc
      ✓ should audit
      ✓ should copy filter values
      ✓ should sort
    clean
      ✓ should replace $ identifiers at BOL
      ✓ should not replace $ identifiers other locations
    sanitize
      ✓ should fail $ identifiers at BOL
[06:11:19] Finished 'json.lint' after 12 s
      ✓ should fail identifiers at BOL with no filters
      ✓ should not replace $ identifiers other locations
    search
      ✓ should perform simple query
      ✓ should perform sorted query
      ✓ should perform limit query
[06:11:19] 
../scaffold/src/web/app/shared/auth/auth.factory.js
  8:2  warning  Unexpected 'todo' comment  no-warning-comments

../scaffold/src/web/app/app.module.js
  38:64  warning  Unexpected 'todo' comment  no-warning-comments

✖ 2 problems (0 errors, 2 warnings)

[06:11:19] Finished 'js.lint' after 13 s
      ✓ should handle query failure

  Mongo reporter
    collection
      ✓ should log entries

  account route
    item
      retrieve
        ✓ should retrieve an account
        ✓ should retrieve an account as admin
        ✓ should fail if not authorized
        ✓ should fail if no account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/account.js:9:834
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
        ✓ should fail on error
      update
        ✓ should update an account
        ✓ should update an account as admin
        ✓ should fail update an account if not authorized
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/account.js:10:6200
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
        ✓ should fail update an account if update fails
    collection
      ✓ should validate accounts
      ✓ should list accounts
      ✓ should list accounts admin
      ✓ should handle empty list
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/account.js:10:2335
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      ✓ should error on list failure
    collection
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/account.js:10:3889
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      1) should create an account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/account.js:10:3889
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      ✓ should error on list failure
    validation
Debug: internal, implementation, error 
    Error: Uncaught error: Configuration property "web.jwtKey" is not defined
    at Config.get (../scaffold/node_modules/config/lib/config.js:178:11)
    at sendValidation (../scaffold/src/server/routes/account.js:9:1250)
    at handler (../scaffold/src/server/routes/account.js:10:7907)
    at Object.exports.execute.internals.prerequisites.internals.handler.callback [as handler] (../scaffold/node_modules/hapi/lib/handler.js:96:36)
    at ../scaffold/node_modules/hapi/lib/handler.js:30:23
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at exports.execute.finalize (../scaffold/node_modules/hapi/lib/handler.js:24:22)
    at each (../scaffold/node_modules/hapi/lib/request.js:383:16)
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:28:25)
    at ../scaffold/node_modules/hapi/lib/protect.js:52:16
    at wrapped (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at Function.wrapped [as _next] (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at Function.internals.Reply.interface.internals.continue (../scaffold/node_modules/hapi/lib/reply.js:107:10)
    at ../scaffold/node_modules/hapi-accept-language/index.js:8:28
    at ../scaffold/node_modules/hapi/lib/request.js:402:22
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at ../scaffold/node_modules/hapi/lib/request.js:397:15
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.callback._protect.run [as _invoke] (../scaffold/node_modules/hapi/lib/request.js:395:19)
      2) should send validation email
      3) should validate
      4) should handle update failure
      5) should handle token error

  audit route
    collection
      ✓ should list audit entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/audit.js:9:1565
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      ✓ should handle db failure 
      ✓ should list audit entries by event/username

  authentication route
    internal
      ✓ should authenticate valid user
      ✓ should authenticate valid user with remember
      6) should refresh token
      7) should refresh token with remember
      ✓ should fail to authenticate invalid user
      ✓ should fail to authenticate on db error
      forgot replace
Debug: internal, implementation, error 
    Error: Uncaught error: Configuration property "web.jwtKey" is not defined
    at Config.get (../scaffold/node_modules/config/lib/config.js:178:11)
    at handler (../scaffold/src/server/routes/authenticate.js:11:2952)
    at Object.exports.execute.internals.prerequisites.internals.handler.callback [as handler] (../scaffold/node_modules/hapi/lib/handler.js:96:36)
    at ../scaffold/node_modules/hapi/lib/handler.js:30:23
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at exports.execute.finalize (../scaffold/node_modules/hapi/lib/handler.js:24:22)
    at each (../scaffold/node_modules/hapi/lib/request.js:383:16)
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:28:25)
    at ../scaffold/node_modules/hapi/lib/protect.js:52:16
    at wrapped (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at Function.wrapped [as _next] (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at Function.internals.Reply.interface.internals.continue (../scaffold/node_modules/hapi/lib/reply.js:107:10)
    at ../scaffold/node_modules/hapi-accept-language/index.js:8:28
    at ../scaffold/node_modules/hapi/lib/request.js:402:22
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at ../scaffold/node_modules/hapi/lib/request.js:397:15
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.callback._protect.run [as _invoke] (../scaffold/node_modules/hapi/lib/request.js:395:19)
    at each (../scaffold/node_modules/hapi/lib/request.js:380:25)
        8) should update password
Debug: internal, implementation, error 
    Error: Uncaught error: Configuration property "web.jwtKey" is not defined
    at Config.get (../scaffold/node_modules/config/lib/config.js:178:11)
    at handler (../scaffold/src/server/routes/authenticate.js:11:2952)
    at Object.exports.execute.internals.prerequisites.internals.handler.callback [as handler] (../scaffold/node_modules/hapi/lib/handler.js:96:36)
    at ../scaffold/node_modules/hapi/lib/handler.js:30:23
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at exports.execute.finalize (../scaffold/node_modules/hapi/lib/handler.js:24:22)
    at each (../scaffold/node_modules/hapi/lib/request.js:383:16)
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:28:25)
    at ../scaffold/node_modules/hapi/lib/protect.js:52:16
    at wrapped (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at Function.wrapped [as _next] (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at Function.internals.Reply.interface.internals.continue (../scaffold/node_modules/hapi/lib/reply.js:107:10)
    at ../scaffold/node_modules/hapi-accept-language/index.js:8:28
    at ../scaffold/node_modules/hapi/lib/request.js:402:22
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at ../scaffold/node_modules/hapi/lib/request.js:397:15
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.callback._protect.run [as _invoke] (../scaffold/node_modules/hapi/lib/request.js:395:19)
    at each (../scaffold/node_modules/hapi/lib/request.js:380:25)
        9) should handle invalid token
Debug: internal, implementation, error 
    Error: Uncaught error: Configuration property "web.jwtKey" is not defined
    at Config.get (../scaffold/node_modules/config/lib/config.js:178:11)
    at handler (../scaffold/src/server/routes/authenticate.js:11:2952)
    at Object.exports.execute.internals.prerequisites.internals.handler.callback [as handler] (../scaffold/node_modules/hapi/lib/handler.js:96:36)
    at ../scaffold/node_modules/hapi/lib/handler.js:30:23
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at exports.execute.finalize (../scaffold/node_modules/hapi/lib/handler.js:24:22)
    at each (../scaffold/node_modules/hapi/lib/request.js:383:16)
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:28:25)
    at ../scaffold/node_modules/hapi/lib/protect.js:52:16
    at wrapped (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at done (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:31:25)
    at Function.wrapped [as _next] (../scaffold/node_modules/hapi/node_modules/hoek/lib/index.js:867:20)
    at Function.internals.Reply.interface.internals.continue (../scaffold/node_modules/hapi/lib/reply.js:107:10)
    at ../scaffold/node_modules/hapi-accept-language/index.js:8:28
    at ../scaffold/node_modules/hapi/lib/request.js:402:22
    at iterate (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:36:13)
    at Object.exports.serial (../scaffold/node_modules/hapi/node_modules/items/lib/index.js:39:9)
    at ../scaffold/node_modules/hapi/lib/request.js:397:15
    at internals.Protect.run.finish [as run] (../scaffold/node_modules/hapi/lib/protect.js:64:5)
    at internals.Request.internals.Request._execute.internals.Request._lifecycle.internals.Request._invoke.callback._protect.run [as _invoke] (../scaffold/node_modules/hapi/lib/request.js:395:19)
    at each (../scaffold/node_modules/hapi/lib/request.js:380:25)
        10) should handle update failure
      forgot email
        ✓ should send forgot password email
        ✓ should ignore forgot password if email fails
        ✓ should ignore forgot password if no user
        ✓ should ignore forgot password on db error

  log route
    collection
      ✓ should list log entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/log.js:9:1562
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      ✓ should handle db failure 
      ✓ should list log entries by event

  metrics route
    collection
      ✓ should retrieve metrics
Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/metrics.js:9:2488
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
      ✓ should handle error

  web route
    ✓ should show index


  52 passing (14s)
  10 failing

  1) account route collection should create an account:

      AssertionError: expected 500 to equal 200
      + expected - actual

      -500
      +200
      
      at test/unit/server/routes/account.js:312:38
      at run (node_modules/core-js/modules/es6.promise.js:89:22)
      at node_modules/core-js/modules/es6.promise.js:102:28
      at flush (node_modules/core-js/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at process._tickDomainCallback (internal/process/next_tick.js:122:9)

  2) account route validation should send validation email:

      AssertionError: expected 500 to equal 200
      + expected - actual

      -500
      +200
      
      at test/unit/server/routes/account.js:362:38
      at run (node_modules/core-js/modules/es6.promise.js:89:22)
      at node_modules/core-js/modules/es6.promise.js:102:28
      at flush (node_modules/core-js/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at Immediate._tickDomainCallback [as _onImmediate] (internal/process/next_tick.js:122:9)

  3) account route validation should validate:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

  4) account route validation should handle update failure:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

  5) account route validation should handle token error:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

  6) authentication route internal should refresh token:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

  7) authentication route internal should refresh token with remember:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

  8) authentication route internal forgot replace should update password:

      AssertionError: expected 500 to equal 200
      + expected - actual

      -500
      +200
      
      at test/unit/server/routes/authenticate.js:85:39
      at run (node_modules/core-js/modules/es6.promise.js:89:22)
      at node_modules/core-js/modules/es6.promise.js:102:28
      at flush (node_modules/core-js/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at Immediate._tickDomainCallback [as _onImmediate] (internal/process/next_tick.js:122:9)

  9) authentication route internal forgot replace should handle invalid token:

      AssertionError: expected 500 to equal 401
      + expected - actual

      -500
      +401
      
      at test/unit/server/routes/authenticate.js:100:39
      at run (node_modules/core-js/modules/es6.promise.js:89:22)
      at node_modules/core-js/modules/es6.promise.js:102:28
      at flush (node_modules/core-js/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at Immediate._tickDomainCallback [as _onImmediate] (internal/process/next_tick.js:122:9)

  10) authentication route internal forgot replace should handle update failure:

      AssertionError: expected 500 to equal 401
      + expected - actual

      -500
      +401
      
      at test/unit/server/routes/authenticate.js:113:39
      at run (node_modules/core-js/modules/es6.promise.js:89:22)
      at node_modules/core-js/modules/es6.promise.js:102:28
      at flush (node_modules/core-js/modules/_microtask.js:18:9)
      at _combinedTickCallback (internal/process/next_tick.js:67:7)
      at Immediate._tickDomainCallback [as _onImmediate] (internal/process/next_tick.js:122:9)



Debug: internal, implementation, error 
    {"message":"10 tests failed.","showStack":true,"showProperties":true,"plugin":"gulp-mocha","__safety":{},"domain":{"domain":null,"_events":{},"_eventsCount":1,"members":[]},"domainThrown":true,"name":"Error"}


```
Scaffold-js: alpha-2
OS: Ubuntu 14.04 LTS
Node version: v5.10.1
NPM version: 3.7.3
YO version: 1.7.0

**Comment by fluky** [comment]
You need to set the JWT_KEY environment variable.  See the section titled "Additional Setup" in the documentation:

https://scaffold-js.github.io/#!/installation

**Comment by m00g33k** [comment]
I didn't choose any social logins though

**Comment by fluky** [comment]
A JWT key is needed no matter what, it is used to sign the tokens so they can be validated as having originated by the server.  The tokens are used for authentication.  Early in development I had generated one using a UUID and stuck it in the config file but as I mentioned on the page above, this would make the private key publicly readable if the generated project were uploaded to github or something.

You can just do:

export JWT_KEY=test
gulp serve

And it should work fine.  I probably should spit out something from the generator at the end to make this more clear.

**Comment by m00g33k** [comment]
Okay, now all the tests are passing, but when I'm trying to connect to localhost:3000 it gives me a browser error "No data recieved ERR_EMPTY_RESPONSE". I get this error also:

```
 Debug: internal, implementation, error 
    Error: Internal Server Error
    at ../scaffold/src/server/routes/metrics.js:9:2488
    at run (../scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at ../scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (../scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
```


> I probably should spit out something from the generator at the end to make this more clear.

It will definitely help, that's for sure. Thank you!

**Comment by fluky** [comment]
There are a couple of odd things about that stack trace.

1.  I'm not sure why it is referring to everything as ../scaffold/....  Are you running gulp serve from the project root or somewhere else?
2.  For me line 9 of src/server/routes/metrics.js is a blank line, and is certainly not 2488 character long.  Can you take a look at that file and make sure it looks normal?
3.  Can you also try bypassing the nodemon/browsersync and just run 'node server' from the project root and see if that makes any difference?

**Comment by m00g33k** [comment]
1. I erased it to make it shorter, but I'll give the full log below
2. Is it maybe because all of it is considered a one liner code? 
3. I tried, but I still cannot connect to the server. Am I missing something?

Anyway, here's the full log
```

[11:19:21] Using gulpfile ~/Workspaces/testFrameworks/scaffold/gulpfile.js
[11:19:21] Starting 'js.lint'...
[11:19:21] Starting 'json.lint'...
[11:19:21] Starting 'test.unit.server.init'...
[11:19:21] Starting 'test.unit.web'...
[11:19:21] Starting 'nm'...
02 05 2016 11:19:22.411:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
02 05 2016 11:19:22.417:INFO [launcher]: Starting browser PhantomJS
[11:19:23] [nodemon] 1.9.2
[11:19:23] [nodemon] to restart at any time, enter `rs`
[11:19:23] [nodemon] watching: src/server/**/*.js
[11:19:23] [nodemon] starting `node server`
[11:19:23] Finished 'nm' after 1.98 s
[11:19:23] Starting 'bs'...
[11:19:23] Finished 'bs' after 43 ms
Server started on port: 8443
[BS] Proxying: http://localhost:8443
[BS] Access URLs:
 ------------------------------------
       Local: http://localhost:3000
    External: http://192.168.1.6:3000
 ------------------------------------
          UI: http://localhost:3001
 UI External: http://192.168.1.6:3001
 ------------------------------------
02 05 2016 11:19:27.258:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket /#p0ITnsGfNBj7cB6aAAAA with id 64998718
PhantomJS 2.1.1 (Linux 0.0.0): Executed 49 of 108 SUCCESS (0 secs / 1.474 secs)
[11:19:30] Finished 'test.unit.server.init' after 8.76 s
PhantomJS 2.1.1 (Linux 0.0.0): Executed 79 of 108 SUCCESS (0 secs / 2.157 secs)


PhantomJS 2.1.1 (Linux 0.0.0): Executed 108 of 108 SUCCESS (2.651 secs / 2.845 secs)
    ✓ should start a server (73ms)
Server started on port: 39036
    ✓ should fail to start a server with an invalid plugin
{ [Error: Cannot find module '/home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/invalid-plugin'] code: 'MODULE_NOT_FOUND' }
    ✓ should fail to start a second server on the same port
[11:19:31] Finished 'test.unit.web' after 11 s

  jwt helpers
    ✓ should validate valid token
    ✓ should fail to validate if host does not match
    ✓ should fail to validate if user not found
    ✓ should fail to validate if db fails

  server methods
    misc
      ✓ should audit
      ✓ should copy filter values
      ✓ should sort
    clean
      ✓ should replace $ identifiers at BOL
      ✓ should not replace $ identifiers other locations
    sanitize
      ✓ should fail $ identifiers at BOL
      ✓ should fail identifiers at BOL with no filters
      ✓ should not replace $ identifiers other locations
    search
[11:19:33] Finished 'json.lint' after 12 s
      ✓ should perform simple query
      ✓ should perform sorted query
      ✓ should perform limit query
      ✓ should handle query failure

  Mongo reporter
[11:19:34] 
/home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js
   7:1  error  Combine this with the previous 'const' statement      one-var
   8:1  error  Combine this with the previous 'const' statement      one-var
   9:1  error  Combine this with the previous 'const' statement      one-var
  10:1  error  Combine this with the previous 'const' statement      one-var
  11:1  error  Combine this with the previous 'const' statement      one-var
  12:1  error  Combine this with the previous 'const' statement      one-var
  21:1  error  Combine this with the previous 'const' statement      one-var
  22:3  error  Expected indentation of 1 tab character but found 2   indent
  41:2  error  Expected indentation of 0 tab characters but found 1  indent
  48:1  error  Combine this with the previous 'const' statement      one-var
  55:1  error  Combine this with the previous 'const' statement      one-var
  56:3  error  Expected indentation of 1 tab character but found 2   indent
  64:3  error  Expected indentation of 1 tab character but found 2   indent
  77:2  error  Expected indentation of 0 tab characters but found 1  indent

/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/shared/auth/auth.factory.js
  8:2  warning  Unexpected 'todo' comment  no-warning-comments

/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/app.module.js
  38:64  warning  Unexpected 'todo' comment  no-warning-comments

✖ 16 problems (14 errors, 2 warnings)

[11:19:34] 'js.lint' errored after 13 s
[11:19:34] ESLintError in plugin 'gulp-eslint'
Message:
    Failed with 14 errors
    collection
      ✓ should log entries

  account route
    item
      retrieve
        ✓ should retrieve an account
        ✓ should retrieve an account as admin
        ✓ should fail if not authorized
        ✓ should fail if no account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:9:1146
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail on error
      update
        ✓ should update an account
        ✓ should update an account as admin
        ✓ should fail update an account if not authorized
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:6200
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail update an account if update fails
    collection
      ✓ should validate accounts
      ✓ should list accounts
      ✓ should list accounts admin
      ✓ should handle empty list
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:2335
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    collection
      ✓ should create an account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:3889
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    validation
      ✓ should send validation email
      ✓ should validate
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:7490
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle update failure
      ✓ should handle token error

  audit route
    collection
      ✓ should list audit entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/audit.js:9:1565
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list audit entries by event/username

  authentication route
    internal
      ✓ should authenticate valid user
      ✓ should authenticate valid user with remember
      ✓ should refresh token
      ✓ should refresh token with remember
      ✓ should fail to authenticate invalid user
      ✓ should fail to authenticate on db error
      forgot replace
        ✓ should update password
        ✓ should handle invalid token
        ✓ should handle update failure
      forgot email
        ✓ should send forgot password email
        ✓ should ignore forgot password if email fails
        ✓ should ignore forgot password if no user
        ✓ should ignore forgot password on db error

  log route
    collection
      ✓ should list log entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/log.js:9:1562
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list log entries by event

  metrics route
    collection
      ✓ should retrieve metrics
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/metrics.js:9:2488
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle error

  web route
    ✓ should show index


  62 passing (4s)

[11:19:36] Finished 'test.unit.server' after 5.94 s
[11:20:04] [nodemon] restarting due to changes...
[11:20:04] [nodemon] starting `node server`
[11:20:04] 'nm' errored after 43 s
[11:20:04] Error: task completion callback called too many times
    at finish (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/orchestrator/lib/runTask.js:15:10)
    at cb (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/orchestrator/lib/runTask.js:29:3)
    at emitNone (events.js:85:20)
    at emit (events.js:179:7)
    at Object.run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/nodemon/lib/monitor/run.js:97:7)
    at ChildProcess.<anonymous> (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/nodemon/lib/monitor/run.js:145:16)
    at emitTwo (events.js:100:13)
    at ChildProcess.emit (events.js:185:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:204:12)
[BS] Reloading Browsers...
Server started on port: 8443
^Cm00g33k@m00g33k-desktop:~/Workspaces/testFrameworks/scaffold$ export JWT_KEY=test; gulp serve
[11:20:10] Using gulpfile ~/Workspaces/testFrameworks/scaffold/gulpfile.js
[11:20:10] Starting 'js.lint'...
[11:20:10] Starting 'json.lint'...
[11:20:10] Starting 'test.unit.server.init'...
[11:20:10] Starting 'test.unit.web'...
[11:20:11] Starting 'nm'...
02 05 2016 11:20:11.400:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
02 05 2016 11:20:11.406:INFO [launcher]: Starting browser PhantomJS
[11:20:12] [nodemon] 1.9.2
[11:20:12] [nodemon] to restart at any time, enter `rs`
[11:20:12] [nodemon] watching: src/server/**/*.js
[11:20:12] [nodemon] starting `node server`
[11:20:12] Finished 'nm' after 1.96 s
[11:20:12] Starting 'bs'...
[11:20:13] Finished 'bs' after 56 ms
Server started on port: 8443
[BS] Proxying: http://localhost:8443
[BS] Access URLs:
 ------------------------------------
       Local: http://localhost:3000
    External: http://192.168.1.6:3000
 ------------------------------------
          UI: http://localhost:3001
 UI External: http://192.168.1.6:3001
 ------------------------------------
02 05 2016 11:20:16.724:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket /#4fjzGvAWDB2fzd0uAAAA with id 38472757
PhantomJS 2.1.1 (Linux 0.0.0): Executed 49 of 108 SUCCESS (0 secs / 1.38 secs)
[11:20:19] Finished 'test.unit.server.init' after 8.93 s
PhantomJS 2.1.1 (Linux 0.0.0): Executed 94 of 108 SUCCESS (0 secs / 2.41 secs)


PhantomJS 2.1.1 (Linux 0.0.0): Executed 108 of 108 SUCCESS (2.739 secs / 2.663 secs)
    ✓ should start a server (79ms)
Server started on port: 56626
    ✓ should fail to start a server with an invalid plugin
{ [Error: Cannot find module '/home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/invalid-plugin'] code: 'MODULE_NOT_FOUND' }
    ✓ should fail to start a second server on the same port

  jwt helpers
[11:20:21] Finished 'test.unit.web' after 11 s
    ✓ should validate valid token
    ✓ should fail to validate if host does not match
    ✓ should fail to validate if user not found
    ✓ should fail to validate if db fails

  server methods
    misc
      ✓ should audit
      ✓ should copy filter values
      ✓ should sort
    clean
      ✓ should replace $ identifiers at BOL
      ✓ should not replace $ identifiers other locations
    sanitize
      ✓ should fail $ identifiers at BOL
      ✓ should fail identifiers at BOL with no filters
      ✓ should not replace $ identifiers other locations
[11:20:23] Finished 'json.lint' after 12 s
    search
      ✓ should perform simple query
      ✓ should perform sorted query
      ✓ should perform limit query
      ✓ should handle query failure

  Mongo reporter
[11:20:23] 
/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/shared/auth/auth.factory.js
  8:2  warning  Unexpected 'todo' comment  no-warning-comments

/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/app.module.js
  38:64  warning  Unexpected 'todo' comment  no-warning-comments

✖ 2 problems (0 errors, 2 warnings)

[11:20:23] Finished 'js.lint' after 13 s
    collection
      ✓ should log entries

  account route
    item
      retrieve
        ✓ should retrieve an account
        ✓ should retrieve an account as admin
        ✓ should fail if not authorized
        ✓ should fail if no account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:9:834
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail on error
      update
        ✓ should update an account
        ✓ should update an account as admin
        ✓ should fail update an account if not authorized
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:6200
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail update an account if update fails
    collection
      ✓ should validate accounts
      ✓ should list accounts
      ✓ should list accounts admin
      ✓ should handle empty list
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:2335
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    collection
      ✓ should create an account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:3889
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    validation
      ✓ should send validation email
      ✓ should validate
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:7490
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle update failure
      ✓ should handle token error

  audit route
    collection
      ✓ should list audit entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/audit.js:9:1565
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list audit entries by event/username

  authentication route
    internal
      ✓ should authenticate valid user
      ✓ should authenticate valid user with remember
      ✓ should refresh token
      ✓ should refresh token with remember
      ✓ should fail to authenticate invalid user
      ✓ should fail to authenticate on db error
      forgot replace
        ✓ should update password
        ✓ should handle invalid token
        ✓ should handle update failure
      forgot email
        ✓ should send forgot password email
        ✓ should ignore forgot password if email fails
        ✓ should ignore forgot password if no user
        ✓ should ignore forgot password on db error

  log route
    collection
      ✓ should list log entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/log.js:9:1562
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list log entries by event

  metrics route
    collection
      ✓ should retrieve metrics
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/metrics.js:9:2488
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle error

  web route
    ✓ should show index


  62 passing (4s)

[11:20:25] Finished 'test.unit.server' after 5.69 s
[11:20:25] Starting 'test.unit'...
------------------------------|----------|----------|----------|----------|----------------|
File                          |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
------------------------------|----------|----------|----------|----------|----------------|
 server/                      |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  jwt.js                      |      100 |      100 |      100 |      100 |                |
  methods.js                  |      100 |      100 |      100 |      100 |                |
 server/models/               |      100 |      100 |      100 |      100 |                |
  account.js                  |      100 |      100 |      100 |      100 |                |
  audit.js                    |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  log.js                      |      100 |      100 |      100 |      100 |                |
  metrics.js                  |      100 |      100 |      100 |      100 |                |
 server/reporters/            |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  mongo.js                    |      100 |      100 |      100 |      100 |                |
 server/routes/               |      100 |      100 |      100 |      100 |                |
  account.js                  |      100 |      100 |      100 |      100 |                |
  audit.js                    |      100 |      100 |      100 |      100 |                |
  authenticate.js             |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  log.js                      |      100 |      100 |      100 |      100 |                |
  metrics.js                  |      100 |      100 |      100 |      100 |                |
  web.js                      |      100 |      100 |      100 |      100 |                |
 web/app/                     |      100 |      100 |      100 |      100 |                |
  app.module.js               |      100 |      100 |      100 |      100 |                |
 web/app/components/account/  |      100 |      100 |      100 |      100 |                |
  account.component.js        |      100 |      100 |      100 |      100 |                |
 web/app/components/apidoc/   |      100 |      100 |      100 |      100 |                |
  apidoc.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/audit/    |      100 |      100 |      100 |      100 |                |
  audit.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/components/forgot/   |      100 |      100 |      100 |      100 |                |
  forgot.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/home/     |      100 |      100 |      100 |      100 |                |
  home.component.js           |      100 |      100 |      100 |      100 |                |
 web/app/components/log/      |      100 |      100 |      100 |      100 |                |
  log.component.js            |      100 |      100 |      100 |      100 |                |
 web/app/components/login/    |      100 |      100 |      100 |      100 |                |
  login.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/components/metrics/  |      100 |      100 |      100 |      100 |                |
  metrics.component.js        |      100 |      100 |      100 |      100 |                |
 web/app/components/register/ |      100 |      100 |      100 |      100 |                |
  register.component.js       |      100 |      100 |      100 |      100 |                |
 web/app/components/topnav/   |      100 |      100 |      100 |      100 |                |
  topnav.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/users/    |      100 |      100 |      100 |      100 |                |
  users.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/shared/account/      |      100 |      100 |      100 |      100 |                |
  account.factory.js          |      100 |      100 |      100 |      100 |                |
 web/app/shared/auth/         |      100 |      100 |      100 |      100 |                |
  auth.directive.js           |      100 |      100 |      100 |      100 |                |
  auth.factory.js             |      100 |      100 |      100 |      100 |                |
 web/app/shared/form/         |      100 |      100 |      100 |      100 |                |
  email.directive.js          |      100 |      100 |      100 |      100 |                |
  match.directive.js          |      100 |      100 |      100 |      100 |                |
  password.directive.js       |      100 |      100 |      100 |      100 |                |
  username.directive.js       |      100 |      100 |      100 |      100 |                |
 web/app/shared/memory/       |      100 |      100 |      100 |      100 |                |
  memory.filter.js            |      100 |      100 |      100 |      100 |                |
 web/app/shared/uptime/       |      100 |      100 |      100 |      100 |                |
  uptime.filter.js            |      100 |      100 |      100 |      100 |                |
------------------------------|----------|----------|----------|----------|----------------|
All files                     |      100 |      100 |      100 |      100 |                |
------------------------------|----------|----------|----------|----------|----------------|

[11:20:25] Finished 'test.unit' after 355 ms
[11:20:25] Starting 'serve'...
[11:20:31] Finished 'serve' after 5.67 s

```

**Comment by m00g33k** [comment]
1.I erased it to make it shorter, but I'll give the full log below
2.Is it maybe because all of it is considered a one liner code?
3.I tried, but I still cannot connect to the server. Am I missing something?
Anyway, here's the full log

```
[11:26:19] Using gulpfile ~/Workspaces/testFrameworks/scaffold/gulpfile.js
[11:26:19] Starting 'js.lint'...
[11:26:19] Starting 'json.lint'...
[11:26:19] Starting 'test.unit.server.init'...
[11:26:19] Starting 'test.unit.web'...
[11:26:19] Starting 'nm'...
02 05 2016 11:26:20.228:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
02 05 2016 11:26:20.234:INFO [launcher]: Starting browser PhantomJS
[11:26:21] [nodemon] 1.9.2
[11:26:21] [nodemon] to restart at any time, enter `rs`
[11:26:21] [nodemon] watching: src/server/**/*.js
[11:26:21] [nodemon] starting `node server`
[11:26:21] Finished 'nm' after 1.95 s
[11:26:21] Starting 'bs'...
[11:26:21] Finished 'bs' after 31 ms
Server started on port: 8443
[BS] Proxying: http://localhost:8443
[BS] Access URLs:
 ------------------------------------
       Local: http://localhost:3000
    External: http://192.168.1.6:3000
 ------------------------------------
          UI: http://localhost:3001
 UI External: http://192.168.1.6:3001
 ------------------------------------
02 05 2016 11:26:25.526:INFO [PhantomJS 2.1.1 (Linux 0.0.0)]: Connected on socket /#XwWZ0kkS0BRtKdifAAAA with id 63725074
PhantomJS 2.1.1 (Linux 0.0.0): Executed 49 of 108 SUCCESS (0 secs / 1.447 secs)
[11:26:28] Finished 'test.unit.server.init' after 8.62 s
PhantomJS 2.1.1 (Linux 0.0.0): Executed 79 of 108 SUCCESS (0 secs / 2.109 secs)


PhantomJS 2.1.1 (Linux 0.0.0): Executed 108 of 108 SUCCESS (2.465 secs / 2.693 secs)
    ✓ should start a server (72ms)
Server started on port: 38977
    ✓ should fail to start a server with an invalid plugin
{ [Error: Cannot find module '/home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/invalid-plugin'] code: 'MODULE_NOT_FOUND' }
    ✓ should fail to start a second server on the same port
[11:26:29] Finished 'test.unit.web' after 10 s

  jwt helpers
    ✓ should validate valid token
    ✓ should fail to validate if host does not match
    ✓ should fail to validate if user not found
    ✓ should fail to validate if db fails

  server methods
    misc
      ✓ should audit
      ✓ should copy filter values
      ✓ should sort
    clean
      ✓ should replace $ identifiers at BOL
      ✓ should not replace $ identifiers other locations
    sanitize
      ✓ should fail $ identifiers at BOL
      ✓ should fail identifiers at BOL with no filters
      ✓ should not replace $ identifiers other locations
[11:26:31] Finished 'json.lint' after 12 s
    search
      ✓ should perform simple query
      ✓ should perform sorted query
      ✓ should perform limit query
[11:26:31] 
/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/shared/auth/auth.factory.js
  8:2  warning  Unexpected 'todo' comment  no-warning-comments

/home/m00g33k/Workspaces/testFrameworks/scaffold/src/web/app/app.module.js
  38:64  warning  Unexpected 'todo' comment  no-warning-comments

✖ 2 problems (0 errors, 2 warnings)

[11:26:31] Finished 'js.lint' after 12 s
      ✓ should handle query failure

  Mongo reporter
    collection
      ✓ should log entries

  account route
    item
      retrieve
        ✓ should retrieve an account
        ✓ should retrieve an account as admin
        ✓ should fail if not authorized
        ✓ should fail if no account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:9:834
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail on error
      update
        ✓ should update an account
        ✓ should update an account as admin
        ✓ should fail update an account if not authorized
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:6200
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
        ✓ should fail update an account if update fails
    collection
      ✓ should validate accounts
      ✓ should list accounts
      ✓ should list accounts admin
      ✓ should handle empty list
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:2335
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    collection
      ✓ should create an account
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:3889
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should error on list failure
    validation
      ✓ should send validation email
      ✓ should validate
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/account.js:10:7490
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle update failure
      ✓ should handle token error

  audit route
    collection
      ✓ should list audit entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/audit.js:9:1565
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list audit entries by event/username

  authentication route
    internal
      ✓ should authenticate valid user
      ✓ should authenticate valid user with remember
      ✓ should refresh token
      ✓ should refresh token with remember
      ✓ should fail to authenticate invalid user
      ✓ should fail to authenticate on db error
      forgot replace
        ✓ should update password
        ✓ should handle invalid token
        ✓ should handle update failure
      forgot email
        ✓ should send forgot password email
        ✓ should ignore forgot password if email fails
        ✓ should ignore forgot password if no user
        ✓ should ignore forgot password on db error

  log route
    collection
      ✓ should list log entries
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/log.js:9:1562
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle db failure 
      ✓ should list log entries by event

  metrics route
    collection
      ✓ should retrieve metrics
Debug: internal, implementation, error 
    Error: Internal Server Error
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/src/server/routes/metrics.js:9:2488
    at run (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:89:22)
    at /home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/es6.promise.js:102:28
    at flush (/home/m00g33k/Workspaces/testFrameworks/scaffold/node_modules/core-js/modules/_microtask.js:18:9)
    at _combinedTickCallback (node.js:370:9)
    at process._tickDomainCallback (node.js:425:11)
      ✓ should handle error

  web route
    ✓ should show index


  62 passing (4s)

[11:26:33] Finished 'test.unit.server' after 5.47 s
[11:26:33] Starting 'test.unit'...
------------------------------|----------|----------|----------|----------|----------------|
File                          |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
------------------------------|----------|----------|----------|----------|----------------|
 server/                      |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  jwt.js                      |      100 |      100 |      100 |      100 |                |
  methods.js                  |      100 |      100 |      100 |      100 |                |
 server/models/               |      100 |      100 |      100 |      100 |                |
  account.js                  |      100 |      100 |      100 |      100 |                |
  audit.js                    |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  log.js                      |      100 |      100 |      100 |      100 |                |
  metrics.js                  |      100 |      100 |      100 |      100 |                |
 server/reporters/            |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  mongo.js                    |      100 |      100 |      100 |      100 |                |
 server/routes/               |      100 |      100 |      100 |      100 |                |
  account.js                  |      100 |      100 |      100 |      100 |                |
  audit.js                    |      100 |      100 |      100 |      100 |                |
  authenticate.js             |      100 |      100 |      100 |      100 |                |
  index.js                    |      100 |      100 |      100 |      100 |                |
  log.js                      |      100 |      100 |      100 |      100 |                |
  metrics.js                  |      100 |      100 |      100 |      100 |                |
  web.js                      |      100 |      100 |      100 |      100 |                |
 web/app/                     |      100 |      100 |      100 |      100 |                |
  app.module.js               |      100 |      100 |      100 |      100 |                |
 web/app/components/account/  |      100 |      100 |      100 |      100 |                |
  account.component.js        |      100 |      100 |      100 |      100 |                |
 web/app/components/apidoc/   |      100 |      100 |      100 |      100 |                |
  apidoc.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/audit/    |      100 |      100 |      100 |      100 |                |
  audit.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/components/forgot/   |      100 |      100 |      100 |      100 |                |
  forgot.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/home/     |      100 |      100 |      100 |      100 |                |
  home.component.js           |      100 |      100 |      100 |      100 |                |
 web/app/components/log/      |      100 |      100 |      100 |      100 |                |
  log.component.js            |      100 |      100 |      100 |      100 |                |
 web/app/components/login/    |      100 |      100 |      100 |      100 |                |
  login.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/components/metrics/  |      100 |      100 |      100 |      100 |                |
  metrics.component.js        |      100 |      100 |      100 |      100 |                |
 web/app/components/register/ |      100 |      100 |      100 |      100 |                |
  register.component.js       |      100 |      100 |      100 |      100 |                |
 web/app/components/topnav/   |      100 |      100 |      100 |      100 |                |
  topnav.component.js         |      100 |      100 |      100 |      100 |                |
 web/app/components/users/    |      100 |      100 |      100 |      100 |                |
  users.component.js          |      100 |      100 |      100 |      100 |                |
 web/app/shared/account/      |      100 |      100 |      100 |      100 |                |
  account.factory.js          |      100 |      100 |      100 |      100 |                |
 web/app/shared/auth/         |      100 |      100 |      100 |      100 |                |
  auth.directive.js           |      100 |      100 |      100 |      100 |                |
  auth.factory.js             |      100 |      100 |      100 |      100 |                |
 web/app/shared/form/         |      100 |      100 |      100 |      100 |                |
  email.directive.js          |      100 |      100 |      100 |      100 |                |
  match.directive.js          |      100 |      100 |      100 |      100 |                |
  password.directive.js       |      100 |      100 |      100 |      100 |                |
  username.directive.js       |      100 |      100 |      100 |      100 |                |
 web/app/shared/memory/       |      100 |      100 |      100 |      100 |                |
  memory.filter.js            |      100 |      100 |      100 |      100 |                |
 web/app/shared/uptime/       |      100 |      100 |      100 |      100 |                |
  uptime.filter.js            |      100 |      100 |      100 |      100 |                |
------------------------------|----------|----------|----------|----------|----------------|
All files                     |      100 |      100 |      100 |      100 |                |
------------------------------|----------|----------|----------|----------|----------------|

[11:26:34] Finished 'test.unit' after 340 ms
[11:26:34] Starting 'serve'...
[11:26:39] Finished 'serve' after 5.56 s

```

**Comment by fluky** [comment]
Ok. That error is from unit tests that are testing unhappy path and are supposed to fail.I don't see any actual errors there.  I'm thinking it is something with Browsersync.  When you start by running "node server" you would run with https://lkcalhost:8443

**Comment by fluky** [comment]
Also, make sure you are connecting with https instead of http and that there is nothing else running on ports 3000, 3991, and 8443,

**Comment by m00g33k** [comment]
Here's what I did:
1. check if somebody is using port 8443, which resulted to none
2. ran "node server" and "http://localhost:8443" which also resulted to none
3. Changed port to 9000 just to double check if its the port, still none.

**Comment by fluky** [comment]
You need to use https not http.

**Comment by m00g33k** [comment]
Okay, that works.

**Comment by m00g33k** [comment]
I believe that browsersync doesn't use https or maybe having problems with it?

**Comment by fluky** [comment]
I'll take a look this evening.  Should work but I haven't used Browserstnc in a while as it tends to slow me down.

**Comment by m00g33k** [comment]
I tried using "gulp serve" and it seems that the server "https://localhost:8443" is working properly.

**Comment by fluky** [comment]
I can't duplicate this here locally.  Are you sure you're using https with browsersync as well?  https://localhost:3000 works great for me.

For the gulp task 'bs', it should be setting the proxy option to 'https://localhost:8443'.  Per the Browsersync doc:  

> https
> 
> Type: Boolean
> Default: undefined
> Note: requires at least version 1.3.0
> 
> Enable https for localhost development. Note - this is not needed for proxy option as it will be inferred from your target url.

As I said it's working for me so hopefully you were just using http://localhost:3000 instead of https.  BTW are you using a remote server?  If you developing locally on a desktop browsersync should automatically open the browser to the correct protocol and port.

**Comment by m00g33k** [comment]
I'm developing locally on a desktop too. And no, the browsersync is not using https when it launches the browser

**Comment by fluky** [comment]
Ok, I think if you update it should probably work.  I had forgotten I had added a fix last night when I looked at the gulpfile.  I published the change and that should take care of browser sync.

**Comment by m00g33k** [comment]
Its working now! Thanks! But I do have a follow up question... How do you properly setup an SSL certificate on a hapijs server? There are alot of tutorials out there but I just want to know which exactly works for scaffold-js

Also, there's a typo on the https://scaffold-js.github.io/#!/development in Generate entity. "yo scaffold-js:**enity**"

**Comment by fluky** [comment]
Thanks for the heads up on the typo, should be fixed now.

As far as certs, the generator creates a snakeoil certificate by default.  It is in the **tls** directory under the root.  Your options are to either replace those with your certificates or to update config/default.js (or override in config/production.js).  Under connections it sets the key and cert files.

The config files use https://github.com/lorenwest/node-config which is a fairly extensive config library.  It loads default.js and if you are running the built version out of dist, it should override anything in the production.js.  If you want to add any additional hapi plugins, this is the place, it uses haps-glue to load all of the plugins from the config.  You can also use environment variables and command line parameters to override, worth checking out the docs for that project.

Thanks for being my guinea pig on this.  So far I'm a one man show, unlike JHipster which has 219 contributors.  I appreciate all feedback and testing.

**Comment by m00g33k** [comment]
There is a Let's encrypt for hapi (https://github.com/Daplie/letsencrypt-hapi) that could be also useful since its a popular choice.

Unfortunately I'm fairly new to NodeJS so I can't contribute that much on the development. But I have a very good intuition and I can predict that this project is capable of growing large as JHipster since this is a very good alternate if you want a lightweight application. To be honest, Hapi is better than express overall and the only problem with hapi is there are less people contributing to it. But I'm hoping that after a beta release, I'm going to help spread this generator and I'm sure that they will put this in the hapijs resource list on their official site. So right now I'll help you to test and use this application to my startup business (it's pretty huge!) as part of my investment on this project.
